### ISSUE TITLE
Concur connector has incorrect data shapes

### ISSUE THREAD

**Issue by zregvart** [syndesisio/syndesis#3041]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem

Swagger file we created Concur connector from has little to do with reality. I just checked with [List Items API Swagger](https://developer.concur.com/api-explorer/v3-0/ListItems.swagger2.json) and the `GET /common/listitems` operation's response is defined as:

```json
    "ListItemGetCollection": {
      "properties": {
        "Items": {
          "$ref": "#/definitions/ListItemGet"
        },
        "NextPage": {
          "type": "string",
          "description": "The URI of the next page of results, if any."
        }
      }
```

And based on the response we receive it should be defined as:

```json
    "ListItemGetCollection": {
      "properties": {
        "Items": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/ListItemGet"
          }
        },
        "NextPage": {
          "type": "string",
          "description": "The URI of the next page of results, if any."
        }
      }
```

One can only safely assume that this mistake was made on all of 30 other Swagger specifications provided and merged into the Concur connector.

## Expected behavior

Data shapes of the connector should converge with reality.

## Screenshot

![ezgif-4-914a9cb720](https://user-images.githubusercontent.com/1306050/42581578-5e5a18d8-852d-11e8-893c-10ca7a5525c9.gif)

**Comment by heiko-braun** [comment]
@zregvart It's not obvious from the description where the differences to the current impl are. Can you provide an example if that?

**Comment by heiko-braun** [comment]
//cc @KurtStam

**Comment by zregvart** [comment]
> @zregvart It's not obvious from the description where the differences to the current impl are. Can you provide an example if that?

Description has JSON schema snippet as defined by the Swagger Concur publishes and what it needs to be.

To make it more plastic, this could be the JSON as defined by the published Swagger:
```json
{
  "Items": { ... }
}
```

Whereas what's actually returned is:

```json
{
  "Items": [
    { ... },
    { ... }
  ]
}
```

**Comment by heiko-braun** [comment]
@KurtStam Is this something you could look into?

**Comment by zregvart** [comment]
This was addressed in #3139
